tests/advent_of_code/test_program_alarm.py

```diff
@@ -0,0 +1,35 @@
+from unittest.mock import patch
+
+
+from code_challenges.advent_of_code.program_alarm import (
+    final_result,
+    get_next_noun_verb_pair,
+    opcode_instructions,
+)
+
+
+@patch("code_challenges.advent_of_code.program_alarm.opcode_integers")
+def test_opcode_instructions(opcode_integers_mock):
+    opcode_integers_mock.return_value = [1, 3, 2, 1, 2, 4, 1, 0, 99, 6, 7]
+    actual = opcode_instructions(4)
+    expected = 0, 0
+
+    assert actual == expected, f"Returned {actual} instead of {expected}"
+
+
+@patch("code_challenges.advent_of_code.program_alarm.opcode_instructions")
+def test_final_result(opcode_instructions_mock):
+    opcode_instructions_mock.return_value = 9, 12
+    actual = final_result()
+    expected = 912
+
+    assert actual == expected, f"Returned {actual} instead of {expected}"
+
+
+def test_get_next_noun_verb_pair():
+    pairs = get_next_noun_verb_pair()
+    next(pairs)
+    next(pairs)
+    next(pairs)
+
+    assert next(pairs) == (0, 3)
```

Comment:
Great, also check that it raises a StopIteration when it should.

Reply:
struggling with this